### ISSUE TITLE
`curses.napms()` aborts with a `SystemError`

### ISSUE THREAD

**Issue by devdanzin** [python/cpython#126313]
# Crash report

### What happened?

The abort happens on normal and no-gil debug builds, release builds also hit the `SystemError`.

```python
import curses
curses.napms(37)
```

Error is:
```
Fatal Python error: _Py_CheckFunctionResult: a function returned a result with an exception set
Python runtime state: initialized
_curses.error: must call initscr() first

The above exception was the direct cause of the following exception:

SystemError: <built-in function napms> returned a result with an exception set

Current thread 0x00007fe41a8c3740 (most recent call first):
  File "/home/fusil/python-73/curses-fatal/source.py", line 3 in <module>
Aborted
```

Found using fusil by @vstinner.

### CPython versions tested on:

3.13, 3.14, CPython main branch

### Operating systems tested on:

Linux

### Output from running 'python -VV' on the command line:

Python 3.14.0a1+ experimental free-threading build (heads/main:d467d9246c, Nov  1 2024, 09:05:56) [GCC 11.4.0]

### Linked PRs
* gh-126351
* gh-126493

**Comment by picnixz** [comment]
Ah I can see what happened. The macro returns 0 so that it can be used as a substitute for NULL in most of the functions it's being used but `_curses.napms` returns an int. Instead, it should return -1. It's an easy fix (one should simply remove the macro usage for this one and do something like that:

```C
 if (!_PyCursesStatefulCheckFunction(module,
								     curses_initscr_called,
									 "initscr")) {
	return -1;
}
```

I think it should suffice. I won't make a PR on Saturday (UTC+2) but probably on Sunday.

**Comment by vstinner** [comment]
Fixed by change https://github.com/python/cpython/commit/19d935843727fff93b2c0e09a63fd32c0d97098d.

**Comment by serhiy-storchaka** [comment]
The fix was not backported yet.

**Comment by picnixz** [comment]
Update: No need for a 3.12 backport because the implementation returns a `PyObject *` instead of an `int`. Only a 3.13 backport is needed (which I am taking care of now).

**Comment by vstinner** [comment]
I merged the 3.13 backport.
